core/Pos/Crypto/HD.hs

```diff
@@ -1,4 +1,129 @@
--- | Hierarchical derivation interface.
+{- Hierarchical derivation interface.
+This module provides basic operations under HD wallets.
+
+* General scheme
+    HD wallet has tree-based structure.
+    The root of the tree is user's public-secret key pair.
+    Each node in the tree corresponds to public-secret key pair.
+
+    Each child node is identified by an index of this child,
+    so child's public and secret keys can be _derived_ from parent
+    by this index using parent's public-secret key pair.
+    On the other hand, a child can't know a parent in reasonable time
+    without some extra information.
+
+* Used structure of tree
+    In general case a structure of tree can be arbitrary but
+    we use specific structure.
+    We have 2 layers tree, nodes of the first layer are called "accounts"
+    and nodes of the second layer are called "addresses" and are leaves of a tree.
+    There can be potentially 2^32 accounts and each account can have 2^32 addresses,
+    so wallet can have 2^64 addresses in total.
+
+    The overall scheme is represented on the picture below.
+
+                                 (wallet's public key and secret key)
+                                  /             |                  \
+                (account[0] (pk, sk))    (account[1] (pk, sk)) ... (account[2^32-1] (pk, sk))
+                /                  \                                    \
+    (address[0][0] (pk, sk)) ... (address[0][2^32-1] (pk, sk)) ... (address[2^32-1][2^32-1] (pk, sk))
+
+    So each address belongs to one account.
+    Addresses are addresses in common sense:
+    they can contain money, money can be spent from them,
+    address can be used as change address and so on.
+
+    Accounts also reflect an intuition behind the word "account":
+    balance of each account can be computed as sum of balances of addresses which belong to it,
+    you can have different accounts for different purposes and so on.
+
+* Derivation process
+    To derive secret key, parent secret key and passphrase are necessary.
+    @deriveHDSecretKey@ does it.
+    Description of underlying cryptography you can get here
```

Comment:
You can get more detailed cryptography specification here:...

Reply:
Also it is wrong because we don't use BIP-32, because it works on secp256k1 only, we use HD/Ed25519 derivation scheme. Here's a spec:
https://cardanolaunch.com/assets/Ed25519_BIP.pdf